pkg/pipeline/resource_pipeline_node_pool.go

```diff
@@ -0,0 +1,212 @@
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// Project GET {{ host }}/access/api/v1/projects/{{prjKey}}/
+//GET {{ host }}/artifactory/api/repositories/?prjKey={{prjKey}}
+type NodePool struct {
+	//Project                   string          `json:"project"`
+	Name                   string   `json:"name"`
+	ProjectId              int      `json:"projectId"`
+	NumberOfNodes          int      `json:"numberOfNodes,omitempty"`
+	IsOnDemand             bool     `json:"isOnDemand"`
+	Architecture           string   `json:"architecture"`
+	OperatingSystem        string   `json:"operatingSystem"`
+	NodeIdleIntervalInMins int      `json:"nodeIdleIntervalInMins"`
+	Environments           []string `json:"environments,omitempty"`
+	ID                     int      `json:"id,omitempty"`
+}
+
+const nodePoolsUrl = "pipelines/api/v1/nodePools"
+
+func pipelineNodePoolResource() *schema.Resource {
+
+	var nodePoolSchema = map[string]*schema.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "The name of the node pool. Should be prefixed with the project key",
+		},
+		"project_id": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+			Description:  "Id of the project where the node pool will live.",
+		},
+		"number_of_nodes": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+			Description:  "Max number of nodes available in the pool.",
+		},
+		"is_on_demand": {
+			Type:        schema.TypeBool,
+			Required:    true,
+			Description: "Set to true for dynamic node pool. Set to false for static node pool.",
+		},
+		"architecture": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
We can do better - Architecture support is an enumeration of strings. Please validate with `validation.StringInSlice` and use the correct architectures.